test/e2e/scrapeconfig_test.go

```diff
@@ -0,0 +1,262 @@
+					{
+						Files: []monitoringv1alpha1.SDFile{
+							"/etc/prometheus/sd/file.json",
+							"/etc/prometheus/sd/file.yaml",
+						},
+						RefreshInterval: &fiveMins,
+					},
+				},
+			},
+		},
+		{
+			name: "invalid-sd-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{
+				FileSDConfigs: []monitoringv1alpha1.FileSDConfig{
+					{
+						Files: []monitoringv1alpha1.SDFile{
+							"/etc/prometheus/sd/file.invalid",
+						},
+						RefreshInterval: &fiveMins,
+					},
+				},
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			testCtx := framework.NewTestCtx(t)
+			defer testCtx.Cleanup(t)
+			ns := framework.CreateNamespace(context.Background(), t, testCtx)
+
+			sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+			sc.Spec = test.spec
+			_, err := framework.MonClientV1alpha1.ScrapeConfigs(ns).Create(context.Background(), sc, metav1.CreateOptions{})
+
+			if test.expectedError {
+				require.Error(t, err)
+				require.Truef(t, apierrors.IsInvalid(err), "expected Invalid error but got %v", err)
+				return
+			}
+			require.NoError(t, err)
+			require.Falsef(t, test.expectedError, "expected error but got nil")
+		})
+	}
+}
+
+// testScrapeConfigLifecycle tests 3 things:
+// 1. Creating a ScrapeConfig and checking that 2 targets appear in Prometheus
+// 2. Updating that ScrapeConfig by adding a target and checking that 3 targets appear in Prometheus
+// 3. Deleting that ScrapeConfig and checking that 0 targets appear in Prometheus
+func testScrapeConfigLifecycle(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, true, true, true, true)
+	require.NoError(t, err)
+
+	p := framework.MakeBasicPrometheus(ns, "prom", "group", 1)
+	p.Spec.ScrapeConfigSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"role": "scrapeconfig",
+		},
+	}
+	_, err = framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	// 1. Create a ScrapeConfig and check that its targets appear in Prometheus
+	sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+	sc.Spec.StaticConfigs = []monitoringv1alpha1.StaticConfig{
+		{
+			Targets: []monitoringv1alpha1.Target{"target1:9090", "target2:9090"},
+		},
+	}
+	_, err = framework.CreateScrapeConfig(context.Background(), ns, sc)
+	require.NoError(t, err)
+
+	// Check that the targets appear in Prometheus
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
+		targets, pollErr := framework.GetActiveTargets(ctx, ns, "prometheus-operated")
```

Comment:
can it use `WaitForActiveTargets` instead?

Reply:
Oh, very nice! 👍  Done in 604686b0611e2b676b49c48e18c4a736d57ac5ca